### ISSUE TITLE
W3 Validator parsing error on CkEditor 4

### ISSUE THREAD

**Issue by madhavonebcg** [ckeditor/ckeditor4#4788]
## Type of report

We are using CKEditor 4 in our web application as a rich text editor. We are compliant with HTML5 standards and guidelines and also providing Accessibility support to our clients in all the projects. There is a tool called [W3 Validator](https://validator.w3.org/) provided by W3C Org to check whether the web applications are following the Markup standards or not. We have tested our application and found that some issues are showed up due to CKEditor as some of the code is not matching W3C standards. The issues are related to the obsolete tags/ attributes used in the markups and also some of the attributes have blank values. So as a loyal and responsible user of CKEditor, we are requesting the devs at CKEditor to fix these issues as soon as possible and when we can expect this to be fixed or even these issues are in the roadmap for development as these issues are becoming the bottleneck the project delivery.

## Provide detailed reproduction steps (if any)
Here are the steps to reproduce the issues in W3 Validator.
1. Visit the web application where CKEditor is used. Open inspect in the browser.
2. Scroll up and find the HTML tag, right-click on it and select the "Edit as HTML" option.
3. Copy all the stuff and head towards [W3 Validator](https://validator.w3.org/).
4. Click on "Validate by Direct Input" and paste the stuff in the text box below and click on the Check button.
5. When results showed up, click on the "Message Filtering" button. Here you can find the issues related to CKEditor

Here are the errors we are getting
- Bad value for attribute src on element iframe: Must be non-empty.
- The ___ attribute on the ___ element is obsolete. Use CSS instead. (2)
        - The frameborder attribute on the iframe element is obsolete. Use CSS instead.
        - The allowtransparency attribute on the iframe element is obsolete. Use CSS instead.

![image](https://user-images.githubusercontent.com/79852850/125042225-3a7f7280-e0b7-11eb-968c-5773f294a63f.png)

### Expected result
Issues should not be showing up on the W3 Validator.

### Actual result
Getting errors on W3 Validator due to CKEditor.

## Other details

* Browser: Google Chrome, Firefox, Safari
* OS: Windows and macOS
* CKEditor version: 4 and above tested

**Comment by Comandeer** [comment]
The violations are connected with two things:

1.  `iframe` source is added via a direct write to its contained document, which results in not changing `iframe`'s `[src]` attribute;
2.  obsolete attributes are added due to [CKEditor's 4 broad browser support](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_browsers.html) – especially old Internet Explorers need them.

Both of these things aren't really fixable. The first one would require changing the whole mechanism of generating editor's editable area only to add a proper `[src]` attribute. The second one is easy to fix, but would result in worse user experience for some of CKEditor 4's users. Due to that I'll close this issue.

Additionally I wonder why do you need fully conformant DOM generated by JS? If it's connected with [WCAG's Success Criterion 4.1.1. Parsing](https://w3c.github.io/wcag/guidelines/22/#parsing), then fully conformance is not needed to comply with it. It requires only syntactical validity of HTML content – and CKEditor 4's DOM is syntactically correct.

**Comment by madhavonebcg** [comment]
Hi @Comandeer, Thanks for your fast response, we will discuss this with clients if they can agree on it. Other than issue this there is another issue I found which left to be mentioned previously.

**Attribute hidefocus not allowed on element a at this point.**

![image](https://user-images.githubusercontent.com/79852850/125072753-99091880-e0d8-11eb-827d-454db4c899f1.png)

Please let me know about this issue as well.
Thanks in advance.

**Comment by Comandeer** [comment]
This issue is very similar to case 2: it's [intended for Internet Explorer](http://help.dottoro.com/lhgdtcso.php).
